fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/domain/SearchQueryRenderer.java

```diff
@@ -2083,10 +2087,16 @@ public void addFilter(QueryData queryData, String resourceType, QueryParameter c
             //                AND P3.PARAMETER_NAME_ID = 123                        -- 'name parameter'
             //                AND P3.STR_VALUE = 'Jones')                           -- 'name filter'
             final int aliasIndex = getNextAliasIndex();
-            final String paramTable = paramValuesTableName(queryData.getResourceType(), currentParm);
             final String paramAlias = getParamAlias(aliasIndex);
-
             WhereFragment pf = paramFilter(currentParm, paramAlias);
+            final String paramTable;
+            if (Type.TOKEN.equals(currentParm.getType()) &&
```

Comment:
If I'm reading the parens correctly here, this could be simplified to TOKEN && TAG && SECURITY